From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: suvojeet-sengupta <suvojeet@x>
Date: Tue, 13 Oct 2026 11:21:39 +0000
Subject: [PATCH] Settings: Move PIF and TEE Simulator into System

They were in security_advanced_settings.xml, which Safety Center does
not surface on calcite, so the entries never showed. Put them in a
Spoofing category on the System dashboard, which is always visible.

Change-Id: If740a1a27a9e1e16fd12a1615704e86ae2227d4f
---
 res/values/pif_strings.xml             |  1 +
 res/xml/security_advanced_settings.xml | 12 ------------
 res/xml/system_dashboard_fragment.xml  | 20 ++++++++++++++++++++
 3 files changed, 21 insertions(+), 12 deletions(-)

diff --git a/res/values/pif_strings.xml b/res/values/pif_strings.xml
index 28fea01..07360cc 100644
--- a/res/values/pif_strings.xml
+++ b/res/values/pif_strings.xml
@@ -32,4 +32,5 @@
     <!-- URL the "Fetch PIF (Penguin server)" button pulls from; empty disables it -->
     <string name="config_pifUpdateUrl" translatable="false"></string>
     <string name="pif_delete_content_description">Clear PIF data</string>
+    <string name="spoofing_category_title">Spoofing</string>
 </resources>
diff --git a/res/xml/security_advanced_settings.xml b/res/xml/security_advanced_settings.xml
index ff186cc..338a131 100644
--- a/res/xml/security_advanced_settings.xml
+++ b/res/xml/security_advanced_settings.xml
@@ -130,19 +130,7 @@
         android:summary="@string/ignore_window_secure_summary"
         settings:controller="com.android.settings.security.IgnoreSecureWindowPreferenceController"/>
 
-    <Preference
-        android:order="306"
-        android:key="tee_simulator"
-        android:title="@string/tee_simulator_title"
-        android:summary="@string/tee_simulator_summary"
-        android:fragment="com.android.settings.custom.spoofing.TeeSimulatorSettings"/>
 
-    <Preference
-        android:order="307"
-        android:key="play_integrity_fix"
-        android:title="@string/play_integrity_fix_title"
-        android:summary="@string/pif_data_summary"
-        android:fragment="com.android.settings.custom.spoofing.PlayIntegrityFix"/>
 
     <PreferenceCategory
         android:order="310"
diff --git a/res/xml/system_dashboard_fragment.xml b/res/xml/system_dashboard_fragment.xml
index 3dae93a..e10b366 100644
--- a/res/xml/system_dashboard_fragment.xml
+++ b/res/xml/system_dashboard_fragment.xml
@@ -180,4 +180,24 @@
             settings:controller="com.android.settings.system.ResetPreferenceController"/>
 
     </PreferenceCategory>
+
+    <PreferenceCategory
+        android:key="spoofing_category"
+        android:title="@string/spoofing_category_title"
+        android:order="-200">
+
+        <Preference
+            android:key="play_integrity_fix"
+            android:title="@string/play_integrity_fix_title"
+            android:summary="@string/pif_data_summary"
+            android:fragment="com.android.settings.custom.spoofing.PlayIntegrityFix"/>
+
+        <Preference
+            android:key="tee_simulator"
+            android:title="@string/tee_simulator_title"
+            android:summary="@string/tee_simulator_summary"
+            android:fragment="com.android.settings.custom.spoofing.TeeSimulatorSettings"/>
+
+    </PreferenceCategory>
+
 </PreferenceScreen>
-- 
2.55.0

